### PR TITLE
Design Updates and Fixes

### PR DESCRIPTION
**Pull request by skipjack**
cc @KyleAMathews @bebraw @oliverturner 

I'm opening this PR to include many of the design changes implemented by @KyleAMathews in #202 as well as updating the [logo](/webpack/media) and applying some other tweaks and fixes to various areas of the site. Here's what's done so far:

- [x] Added Geomanist and Cabin
- [x] Cleaned up the layout a bit
- [x] Simplified color scheme using @jhnns logo colors
- [ ] Sort out the license and bring @jhnns' webpack/media work in
- [x] Bring the mobile sidebar back to life
- [x] Finish any other design tweaks
- [x] Add favicon

@KyleAMathews let me know if you think I missed anything design-wise. I think we want to stay away from js/inline styles, at least for now, and stick with the modular-scale plugin. Thanks again for your work on #202.

@jhnns @bebraw hoping we can get the license sorted out here but I'm not really sure how to handle this.

**Note:** Please be aware that right now certain features will only work the dev-server but before we get this merged I'm hoping to have things fully functional (with @bebraw's `interactive` addition I believe we can).

### CODE REVIEW

File: styles/index.scss [hunk 1]
```diff
@@ -24,9 +21,8 @@ html {
 }
 
 body {
-  font: 300 getFontSize(0) 'Open Sans', 'Century Gothic', sans-serif;
+  font: 300 getFontSize(0) 'Cabin', 'Century Gothic', sans-serif;
```
Comment (KyleAMathews) [inline styles/index.scss]: I'd suggest using 400 for body text not 300. < 400 gets a bit hard to read on many devices.

**Comment by KyleAMathews** [comment]
I haven't tested it but can't see why react-headroom wouldn't work with Preact.

File: styles/index.scss (hunk 1 above)
Comment (gregvenech) [inline styles/index.scss]: Good catch, I was actually only importing Cabin in 400 and 600 (with italic versions of each) but forgot to update this... changing momentarily.

**Comment by gregvenech** [comment]
@KyleAMathews yea I just noticed when `npm install`ing `react-headroom` that a react addon was installed as well and I wasn't sure if that would play well with `preact`. I'm not too familiar with `preact` yet, or `react-headroom` for that matter but looking forward to playing around with both more.

**Comment by gregvenech** [comment]
Ok pretty much finished up here... the Sidebar still needs some work but I think we can open another PR for that. At a basic level I have it working on all viewports even with the `react-sticky` addition (though it did take some hacks 😉 ).

@bebraw somehow I broke the `Interactive` behavior, everything runs fine on the dev-server but when I do a build and test the `/build` folder with `http-server` the interactive components are broken. I think it has to do with `react-sticky` and `react-headroom` dependencies. If you could take a look that'd be awesome. If it is those packages we can take them out for now as both are helpful but not critical.

@jhnns added the new logo/icon to the Navigation and Footer. Feel free to make any tweaks, in the long run it would be nice to use a component rather than the static image but for now I just wanted to get them in. Maybe if we just add something to the license like:

> Except where otherwise stated, this code is:
> ...
> The logo and icons images are under a different license which can be found [here](/webpack/media).

Once we get these bits sorted out I think we should be good to merge this.

**Comment by bebraw** [comment]
@gregvenech I can see

```bash
Module not found: Error: Cannot resolve module 'preact-compat/lib/ReactComponentWithPureRenderMixin' in .../webpack.js.org/node_modules/react-addons-pure-render-mixin
```

while building. The problem is that `preact-compat` doesn't implement it. Most likely you'll have to fork the components in question and adjust them so that they work with Preact.

**Comment by KyleAMathews** [comment]
If there's a Preact friendly way to make a component pure, I'd take a PR for that to react-headroom.

**Comment by bebraw** [comment]
@KyleAMathews Gotcha. I'll ask around a bit. 👍

**Comment by bebraw** [comment]
@KyleAMathews `preact-compat` 3.6.0 has `PureComponent`. See https://github.com/developit/preact-compat/releases/tag/3.6.0 . So I guess that would be the way.

**Comment by gregvenech** [comment]
Ok so this PR is pretty much ready from my standpoint, I removed react-headroom and react-sticky for now but these can be re-opened in separate PRs (these can be started by simply reverting my "Removing react-..." commits).

@bebraw imported image paths are still broken for some reason, it seems like they shouldn't be though - see 0dcf7a5. I noticed that even some of the fonts seem to have issues and get copied both into `/assets` as well as the root.

Other than that just waiting on the license change. I'm going to open some issues for the things that still need work so we don't forget.

**Comment by TheLarkInn** [comment]
@gregvenech is the license change PR the one referenced above? Would you mind linking it in your latest comment.

**Comment by gregvenech** [comment]
@TheLarkInn I didn't see a license PR in this repo just the [one over at wepack media](https://github.com/webpack/media/pull/3).

File: components/page/page-style.scss [hunk 2]
```diff
@@ -16,7 +16,7 @@
 .page__content {
   overflow-x: hidden;
   width: 100%;
-  padding: 0.5em;
+  padding: 1.5em 1em;
 
   @media (min-width: map-get($screens, medium)) {
```
Comment (oliverturner) [inline components/page/page-style.scss]: any reason not to simply use 
```scss
.page__content {
  ...
  @include break {
    ...
  }
}
```

Comment (gregvenech) [inline components/page/page-style.scss]: Good point, fixing now.

**Comment by SpaceK33z** [comment]
@gregvenech, are you going to make big changes or is this almost ready? Asking because I want to test this out, but that it would be a waste of time if more changes are coming ;).

**Comment by gregvenech** [comment]
@SpaceK33z nope in my mind this branch is good to go. I want to push any other changes to new PRs (e.g. more sidebar work, landing page, etc.). The only problems currently with merging are:

1. Not sure if we want to merge this before the license is tweaked.
2. Everything works fine on the dev-server but images aren't working in production builds. Waiting for @bebraw to take a look.

Aside from that it's good to go and I think we should get it merged asap.

**Comment by TheLarkInn** [comment]
Before we merge could we make one more change? That would involve adding our Open Collective banner. It's done via script tag. https://github.com/OpenCollective/OpenCollective/wiki/Website-banner

Would you be able to add this, if it trivial, if it is not, we can separate this into a separate PR and Issue and then merge this and cut a release.

**Comment by gregvenech** [comment]
@TheLarkInn yea I can try tonight. I'm thinking maybe just on the landing page?

**Comment by TheLarkInn** [comment]
Well you know what, let me put it in a separate issue. If this is ready to go lets merge it.

**Comment by TheLarkInn** [comment]
Added to separate issues #248 and we'll do a separate PR for it.

**Comment by gregvenech** [comment]
@TheLarkInn sounds good, yea I mean if you guys are comfortable merging it regardless of those two things I mentioned then by all means. We could also tweak the license quickly for now by adding the language I mentioned before...

> Except where otherwise stated, this code is:
> ...
> The logo and icons images are under a different license which can be found [here](/webpack/media).

And come back to it later if necessary. I'm no expert with licensing so I was kind of defaulting for all of your suggestions on this.

**Comment by TheLarkInn** [comment]
I'm comfortable with this. @SpaceK33z is doing to test later and then we can merge.

**Comment by bebraw** [comment]
@gregvenech About assets, fastest fix might be to copy images/fonts below `assets` directly and then fix paths to point there. In this case you'll end up with a single copy clause.

Ideally asset pipeline should go through webpack entirely, but that's something to tackle separately.

**Comment by gregvenech** [comment]
@bebraw @TheLarkInn sounds good 👍 . I will make the changes to the license and try the assets suggestions.

**Comment by SpaceK33z** [comment]
Tested it:

- Could the header be more like Kyle's design? The larger font + active state are a lot better in that design.
- Kyle's design has a nice active state in the sidebar. You previously said you're going to do more work on the sidebar, so will you also take this into account then?
- The "Edit this page" link is blue, maybe change this to be the same as other links?
- The headings are the same color as the body text. Is this on purpose? In Kyle's design they are darker, so it stands out more.
- Woah the footer looks nice now!

I also asked a designer-friend (@indri-indri) to look at this, he suggested this enhancement:

**Before:**

webpack

**After** adding `text-rendering: geometricPrecision;` to `.logo__text`:

webpack

**Comment by gregvenech** [comment]
1. Yea I can tweak the header, personally I like it more subtle but I see what you mean.
2. Exactly, I would push this to a separate PR, I can add this note to that issue. I leaning towards a separate MobileSidebar component as @KyleAMathews had instead of using the current one for both.
3. I've gone back and forth on that edit link, and I have tried it both ways. Problem was it wasn't popping enough in that dark yellow color. Imo this link should stand out from the others.
4. Yep I can definitely tweak the header color.

Will jump on these things and the two others in a few hours.

**Comment by skipjack** [comment]
Ok should be ready for another review. There's only one image path broken in the generated `/build` which can either be fixed manually when deploying or by @bebraw once he has time. Dev server is still good and all discussed changes have been added.

**Comment by jhnns** [comment]
Regarding the logo in the header: What's your opinion on changing the header to the actual image (instead of using HTML text)? Font rendering can be very shitty depending on the OS, so as a designer I would definitely prefer to use the pre-rendered image for the banner :)

**Comment by SpaceK33z** [comment]
@jhnns, you mean to add the text as paths instead of the real font, right? That would definitely be better.

**Pull request merged**